_posts/2016-05-16-rust-at-one-year.md

```diff
@@ -0,0 +1,269 @@
+
+[MSVC toolchain]: https://github.com/rust-lang/rust/pull/25350
+[MUSL]: https://www.musl-libc.org/
+[platforms]: https://forge.rust-lang.org/platform-support.html
+[rustup tool]: http://blog.rust-lang.org/2016/05/13/rustup.html
+[no_std]: http://blog.rust-lang.org/2016/01/21/Rust-1.6.html
+[intermezzos]: https://intermezzos.github.io/
+[neon]: http://calculist.org/blog/2015/12/23/neon-node-rust/
+[helix]: http://blog.skylight.io/introducing-helix/
+[rure-go]: https://github.com/BurntSushi/rure-go
+
+- **Tools**. Because Rust looks just like C on the outside, it's instantly
+  usable with a wide range of existing tools; it works out of the box with
+  [lldb], [gdb], [perf], [valgrind], [callgrind], and many, many more.  Our
+  focus has been to [enrich the experience] for these tools by adding
+  [Rust-specific hooks][gdb hooks] and [workflows][cargo profile]. Another major
+  priority is providing full IDE support, in part by providing daemonized
+  services from the compiler; we made [good progress][IDEs] on that front this
+  year, and thanks to the [Racer] project, [numerous IDE plugins] are already
+  providing some semantic support for Rust. At the same time, the [rustfmt] code
+  formatting tool has matured to the point that the Rust community is ready to
+  produce an [official style].  And the beating heart of Rust's workflow,
+  [Cargo], gained numerous abilities this year, most notably the
+  [install subcommand].
+
+[lldb]: http://lldb.llvm.org/
+[gdb]: https://www.gnu.org/software/gdb/
+[perf]: https://perf.wiki.kernel.org/index.php/Main_Page
+[valgrind]: http://valgrind.org/
+[callgrind]: https://kcachegrind.github.io/html/Home.html
+[enrich the experience]: https://michaelwoerister.github.io/2015/03/27/rust-xxdb.html
+[gdb hooks]: https://sourceware.org/ml/gdb-patches/2016-04/msg00570.html
+[cargo profile]: http://www.suchin.co/2016/05/11/Introducing-Cargo-Profiler/
+[IDEs]: https://www.rust-lang.org/ides.html
+[Racer]: https://github.com/phildawes/racer
+[numerous IDE plugins]: https://areweideyet.com/
+[rustfmt]: https://github.com/rust-lang-nursery/rustfmt
+[official style]: https://github.com/rust-lang/rfcs/pull/1607
+[Cargo]: http://blog.rust-lang.org/2016/05/05/cargo-pillars.html
+[install subcommand]: http://blog.rust-lang.org/2015/12/10/Rust-1.5.html
+
+- **Compiler**. We've seen some [across-the-board improvements] to compile
+  times, and now offer [parallelized code generation][parallel codegen] for
+  further speedups. But the biggest wins will come from the ongoing work on
+  [incremental compilation], which will minimize the amount of work the needed
+  when recompiling code after editing it. A vital step here was the move to a
+  [custom intermediate representation][MIR], which has many other benefits as
+  well. Another focus has been errors, including
+  [detailed explanations of most errors][error index], and ongoing work to
+  [improve the clarity and focus of errors][error format]. Expect to hear more
```

Comment:
Maybe instead of "improve the clarity and focus of errors", can we say something like "improve the 'at a glance' readability of errors" or something like that?  Clarity and focus is helpful, but our goal is on the developer understanding the error as fast as possible.